tests/containers/rootless_podman.pm

```diff
@@ -0,0 +1,64 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test rootless mode on podman.
+# - add a user on the /etc/subuid and /etc/subgid to allow automatically allocation subuid and subgid ranges.
+# - check uids allocated to user (inside the container are mapped on the host)
+# - give read access to the SUSE Customer Center credentials to call zypper from in the container
+#   proper way to add Access Control List is via `setfacl` which is not available so we just do
+#   `chmod` instead!! This grants the current user the required access rights
+# - Test rootless container:
+#   * container is launched with default root user
+#   * container is launched with existing user id
+#   * container is launched with keep-id of the user who run the container
+# - Restore /etc/zypp/credentials.d/ credentials
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use testapi;
+use utils;
+use containers::common;
+use containers::container_images;
+use suse_container_urls 'get_suse_container_urls';
+use version_utils qw(get_os_release);
+
+sub run {
+    my ($self) = @_;
+    my ($image_names, $stable_names) = get_suse_container_urls();
+    my ($running_version, $sp, $host_distri) = get_os_release;
+    my $runtime = "podman";
+
+    install_podman_when_needed($host_distri);
+    allow_selected_insecure_registries(runtime => $runtime);
+    my $user = $testapi::username;
+    assert_script_run "cat /etc/subuid | grep bernhard || echo /etc/subuid has no uid range for $user";
+    assert_script_run "cat /etc/subgid | grep bernhard || echo /etc/subgid has no gid range for $user";
+    assert_script_run "usermod --add-subuids 200000-201000 --add-subgids 200000-201000 $user";
```

Comment:
Why does this add a range if it just verified if it already has one?

Reply:
looks like your comment proving mine comment above :) there is no **verification** meaning that there is no assertion so in there is no range test will continue running

Reply:
The range defined should not overlap for each user. i just make sure that there are not configuration for the current user.

Reply:
lines 40 and 41 are just informational now , you will run `usermod` in any case . maybe better to split this line into two and tight execution to 40 and 41. Something like this ( just raw pseudo code needs to be checked ) : 
```
unless (script_run ( 'grep -q bernhard /etc/subuid' ) ) {
     assert_script_run "usermod --add-subuids 200000-201000"
} 
unless (script_run ( 'grep -q bernhard /etc/subgid' ) ) {
     assert_script_run "usermod --add-subgids 200000-201000"
} 
```

Reply:
or even better :) 

```
assert_script_run "grep -q bernhard /etc/subuid || usermod --add-subuids 200000-201000"
assert_script_run "grep -q bernhard /etc/subgid || usermod --add-subgids 200000-201000"
```

Reply:
@asmorodskyi i tried but it didnt work for now. promise to check another time

Reply:
I must admit that this looks weird to me as well. Can you investigate this further, Yiannis and or at least provide us with some more details?

Reply:
more details on what? the greps? i am not sure what else to say beyond the ones already said

Reply:
please check the documentation https://documentation.suse.com/sles/15-SP2/single-html/SLES-container/index.html#cha-podman-install

Reply:
thanks a lot for this URL it really helps